static/styles/zulip.css

```diff
@@ -2758,6 +2758,15 @@ select.inline_select_topic_edit {
     }
 }
 
+.placeholder {
+    color: hsl(0, 0%, 66.3%);
+    opacity: 1;
+}
+
+::placeholder {
+    @extend .placeholder;
+}
```

Comment:
I made several changes here:
* Added a block comment explaining the motivation, since it seems worth having here for posterity.
* Made sure bootstrap wasn't overriding this CSS, which it was in the light theme
* Changed the color to 45% in the day theme and 55% in the night theme, to meet WCAG accessibility standards (I verified that the GitHub reply button does this).